### PR TITLE
fix(autocomplete): don't scroll panel when option is visible

### PR DESCRIPTION
**Pull request by willshowell**
When scrolling down to an option well below the fold and then scrolling back up a couple of options, it feels jarring to have the selected option "stick" to the bottom of the panel. This fix keeps the panel from scrolling when the selected option is fully visible.

cc @kara 

-----

**Before**
autocomplete-before

**After**
autocomplete-after

### CODE REVIEW

**Comment by willshowell** [comment]
@jelbourn could I get this assigned? 😄

**Comment by willshowell** [comment]
@kara I would really appreciate if you could review this! I think it makes keyboard scrolling feel more natural.

File: src/lib/autocomplete/autocomplete.spec.ts [hunk 1]
```diff
@@ -763,6 +763,64 @@ describe('MdAutocomplete', () => {
...
+
+      // These up arrows will set the 2nd option active
+      [4, 3, 2, 1].forEach(() => {
+        fixture.componentInstance.trigger._handleKeydown(UP_ARROW_EVENT);
+        tick();
+      });
+
+      // Expect no scrolling to have occurred. Still showing bottom of 6th option.
+      expect(scrollContainer.scrollTop)
+        .toEqual(32, `Expected panel to not scroll back.`);
```
Comment (kara) [inline src/lib/autocomplete/autocomplete.spec.ts]: Nit: The failure message is a bit vague. It seems like you should move the guidance in the comment to the failure message.

**Comment by willshowell** [comment]
@kara changes made!

**Pull request merged**